universe/Universe.cpp

```diff
@@ -2209,77 +2209,82 @@ void Universe::ExecuteEffects(const Effect::TargetsCauses& targets_causes,
 
             if (effects_group != last_effects_group) {
                 last_effects_group = effects_group;
-                dispatched_targets_causes.push_back(std::make_pair(effects_group, Effect::TargetsCauses()));
-                group_targets_causes = &dispatched_targets_causes.back().second;
+                dispatched_targets_causes[effects_group->Priority()].push_back(std::make_pair(effects_group, Effect::TargetsCauses()));
+                group_targets_causes = &dispatched_targets_causes[effects_group->Priority()].back().second;
             }
             group_targets_causes->push_back(*targets_it);
         }
     }
 
     // execute each effects group one by one
-    std::vector<std::pair<Effect::EffectsGroup*, Effect::TargetsCauses> >::iterator effect_group_it;
-    for (effect_group_it = dispatched_targets_causes.begin();
-         effect_group_it != dispatched_targets_causes.end(); ++effect_group_it)
+    std::map<int, std::vector<std::pair<Effect::EffectsGroup*, Effect::TargetsCauses> > >::reverse_iterator priority_group_it;
+    for (priority_group_it = dispatched_targets_causes.rbegin();
```

Comment:
Depends on your definition of "least" and "greater", but if it just means "smallest" and "biggest", an earlier effect that reduces a meter from 10 to 5 could be said to have a much bigger (changed by absolute amount of 5) or much smaller (changed by -5) effect than a later effect that adds 50% (2.5) to the meter.

Point being the terms used are meaningless. Defaulting to priority 0 when higher priorities go first is probably fine, though.